### PR TITLE
[PS-248] Feature/manage billing sync connection

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds api response/request for Organization connections

## Code changes
* **api.service**: CRUD for organization connections. I had to use a abstract method pattern vs our typical (and weird) property pattern to allow for generic arguments on the methods. These generics are used to correctly parse the config.
* **organizationConnectionType**: enum for the connection type.
* **billingSyncConfigApi**: the api response for billingSyncConfigs
* **billingSyncConfigRequest**: The api request model for billing sync config
* **OrganizationConnecitonRequest**: The api request model for general organization connections. Config types are limited by the `OrganizationConnectionRequestConfigs` type
* **organizationConnectionResponse**: The api response model for general organization connections. Config types are limited by the `OrganizationConnectionConfigApis` type

## Testing requirements

N/A Testing will be done with Web client.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: common/src/models/request/organizationConnectionRequest.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export class OrganizationConnectionRequest {
+  config: string;
+
+  constructor(
+    public organizationId: string,
+    public type: OrganizationConnectionType,
+    public enabled: boolean,
+    config: OrganizationConnectionRequestConfigs
+  ) {
+    this.config = JSON.stringify(config);
```
Comment (justindbaur) [inline common/src/models/request/organizationConnectionRequest.ts:17]: Do we need this anymore with the server changes?

File: common/src/models/response/organizationConnectionResponse.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { OrganizationConnectionType } from "jslib-common/enums/organizationConnectionType";
+
+import { BillingSyncConfigApi } from "../api/billingSyncConfigApi";
+
+import { BaseResponse } from "./baseResponse";
+import { NotificationResponse } from "./notificationResponse";
+
+/**API response config types for OrganizationConnectionResponse */
+export type OrganizationConnectionConfigApis = BillingSyncConfigApi | NotificationResponse;
```
Comment (justindbaur) [inline common/src/models/response/organizationConnectionResponse.ts:9]: What is `NotificationResponse`?

Comment (MGibson1) [reply common/src/models/response/organizationConnectionResponse.ts:9]: oops, thank you, I was testing out the compiler hints with a random response and forgot to delete this 😊

File: common/src/models/response/organizationConnectionResponse.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { OrganizationConnectionType } from "jslib-common/enums/organizationConnectionType";
+
+import { BillingSyncConfigApi } from "../api/billingSyncConfigApi";
+
+import { BaseResponse } from "./baseResponse";
+import { NotificationResponse } from "./notificationResponse";
```
Comment (justindbaur) [inline common/src/models/response/organizationConnectionResponse.ts:6]: Just need to remove this.

**Review by justindbaur — CHANGES_REQUESTED** [review]
Just fix the linter error and then good to go.

**Review by justindbaur — APPROVED** [review]

**Pull request merged**